pkg/engine/engine.go

```diff
@@ -426,6 +428,67 @@ func (e *Engine) Kill(id string) error {
 	return nil
 }
 
+// CleanUpTasks deletes all tasks in the store that are currently being processed
+func (e *Engine) CleanUpTasks() error {
+
+	logging.S().Infow("Sleeping for a while, allowing sync service to start...")
+	time.Sleep(10 * time.Second)
+	logging.S().Infow("Waking up")
+	arunner, success := e.RunnerByName("cluster:k8s")
+
+	if !success {
+		return errors.New("no cluster k8s runner")
+	}
+
+	krunner := *arunner.(*runner.ClusterK8sRunner)
+
+	allTasks, err := e.store.AllTasks()
+	if err != nil {
+		return err
+	}
+
+	for _, aTask := range allTasks {
+
+		// TODO: parse the configuration (runInput/runnerconfig)  from the task, then pass it along to the runner
+		if aTask.State().State != task.StateScheduled {
+			// If the engine has a signal for the task, that means it's being tracked actively
+			if e.hasSignal(aTask.ID) {
+				logging.S().Infof("Skipping cleanup of task %s, as it is still active: %s", aTask.ID, aTask.State())
+				continue
+			}
+
+			if aTask.State().State == task.StateProcessing {
+				err = e.store.ArchiveTask(aTask)
+				if err != nil {
+					logging.S().Warnf("Error archiving task %s: %s", aTask.ID, err.Error())
+				}
+			}
+
+			var pods *v1.PodList
+			if len(aTask.ID) > 0 {
+				pods, err = krunner.ListAllPods(context.TODO(), aTask.ID)
+				if err != nil {
+					return err
+				}
+			}
+
+			if len(pods.Items) > 0 {
+				logging.S().Infof("Cleaning up inactive task %s with %d pods", aTask.ID, len(pods.Items))
+				for _, pod := range pods.Items {
+					logging.S().Infof("Deleting pod %s", pod.Name)
+					err = krunner.DeleteWorkPod(pod.Name, aTask.Input)
+					if err != nil {
+						return fmt.Errorf("error deleting pod %s: %v", pod.Name, err)
+					}
+				}
+			}
+		}
+	}
+
+	return nil
+
+}
```

Comment:
That code looks very specific to kubernetes, we should stick to the `engine` abstraction and have something like

```
for every scheduled tasks:
   runner[task.runner].cleanup()
```

Then implement the cleanup in each runner (`docker` and `k8s`)